backend/director/agents/download.py

```diff
@@ -30,13 +28,33 @@ def run(
         :rtype: AgentResponse
         """
         try:
+            text_content = TextContent(agent_name=self.agent_name)
+            text_content.status_message = "Downloading.."
+            self.output_message.content.append(text_content)
+            self.output_message.push_update()
             videodb_tool = VideoDBTool()
             download_response = videodb_tool.download(stream_link, name)
+            if download_response.get("status") == "done":
+                download_url = download_response.get("download_url")
+                name = download_response.get("name")
+                text_content.text = (
+                    f"<a href='{download_url}' target='_blank'>{name}</a>"
+                )
+                text_content.status = MsgStatus.success
+                text_content.status_message = "Here is the download link"
+                self.output_message.publish()
+            else:
```

Comment:
Can we also change the `text_content` to error before returning error `AgentResponse` ?

Reply:
Pushed, can you test that condition now? it should set the error state

Reply:
yeah, working now.

<img width="1292" alt="Screenshot 2024-11-25 at 11 48 42 AM" src="https://github.com/user-attachments/assets/b4e0a945-b918-4bda-aad5-b2c0cc63b2e2">